packages/lodestar/src/network/gossip/gossipsub.ts

```diff
@@ -89,15 +49,67 @@ export class LodestarGossipsub extends Gossipsub {
     const subscribedTopics = super.getTopics();
     topics.forEach((topic) => {
       if (subscribedTopics.includes(topic)) {
-        const transformedObj = this.transformedMessages.get(this.getKey(message));
+        const transformedObj = this.transformedObjects.get(this.getKey(message));
         if (transformedObj) {
           super.emit(topic, transformedObj);
-          this.transformedMessages.delete(this.getKey(message));
+          this.transformedObjects.delete(this.getKey(message));
         }
       }
     });
   }
 
+
+  private getTopicValidator(topic: string): GossipMessageValidatorFn {
+    for(let subnet = 0; subnet < ATTESTATION_SUBNET_COUNT; subnet++) {
+      if (topic === getGossipTopic(GossipEvent.ATTESTATION_SUBNET, "ssz", new Map([["subnet", String(subnet)]]))) {
+        return this.validator.isValidIncomingCommitteeAttestation as GossipMessageValidatorFn;
+      }
+    }
+
+    switch(topic) {
+      case getGossipTopic(GossipEvent.BLOCK, "ssz"):
+        return this.validator.isValidIncomingBlock as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.ATTESTATION, "ssz"):
+        return this.validator.isValidIncomingUnaggregatedAttestation as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.AGGREGATE_AND_PROOF, "ssz"):
+        return this.validator.isValidIncomingAggregateAndProof as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.ATTESTER_SLASHING, "ssz"):
+        return this.validator.isValidIncomingAttesterSlashing as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.PROPOSER_SLASHING, "ssz"):
+        return this.validator.isValidIncomingProposerSlashing as GossipMessageValidatorFn;
+      case getGossipTopic(GossipEvent.VOLUNTARY_EXIT, "ssz"):
+        return this.validator.isValidIncomingVoluntaryExit as GossipMessageValidatorFn;
```

Comment:
I think codeclimate will complain about too many returns. You could assign that fn to variable and cast only once as well :)